cogito/ghcommitsink.go

```diff
@@ -54,6 +54,9 @@ func (sink GitHubCommitStatusSink) Send() error {
 		"state", ghState, "owner", sink.Request.Source.Owner,
 		"repo", sink.Request.Source.Repo, "git-ref", sink.GitRef,
 		"context", context, "buildURL", buildURL, "description", description)
+	if sink.Request.Source.OmitTargetURL {
+		buildURL = ""
```

Comment:
As the GH API expect a string or null, wouldn't be cleaner to pass null instead?
(https://docs.github.com/en/rest/commits/statuses?apiVersion=2022-11-28#create-a-commit-status)

Reply:
I'm also tempted to do this before logging to avoid a misleading `buildURL` showing up in the logs

Reply:
> I'm also tempted to do this before logging to avoid a misleading `buildURL` showing up in the logs

That was my first iteration. Then I realized that the logging should still contain the buildURL, since the logging is not publicly accessible, and the goal of this PR is to omit the Concourse URL from a public GitHub PR page. Omitting also from the log would, in my understanding, make troubleshooting or auditing more difficult for no reason...

Reply:
> As the GH API expect a string or null, wouldn't it be cleaner to pass null instead? (https://docs.github.com/en/rest/commits/statuses?apiVersion=2022-11-28#create-a-commit-status)

The type of buildURL is a string, not a pointer to a string, so we cannot set it to nil. We could change the type to pointer to string, but that would bring pain everywhere else.

What we could do is to decorate the structure field `TargetURL` with tag `omitempty`

https://github.com/Pix4D/cogito/blob/a55a8121dbcc3b9ad96c573d0a37dcfc93bdf8f7/github/commitstatus.go#L83-L89

like this:

```
	TargetURL   string `json:"target_url,omitempty"`
```

`omitempty` will see an empty string (the value `""`) and will omit the field from the JSON object. This is not the same as setting the field to JSON null...

To summarize, I am tempted to say that if the GH API is happy with an empty string, that is good enough. But this discussion made me realize that I need to test this somehow.

Reply:
@odormond I am mentioning you here because I am going to merge this PR. I believe I addressed both your comments. If I missed something, ping me.